### PR TITLE
[BUG] fixing multi-hit and move messages on faint

### PR DESCRIPTION
**Pull request by DustinLin**
## What are the changes?

the same as PR #2857 , but that was purged so reopening

### CODE REVIEW

**Review by CodeTappert — APPROVED** [review]

**Comment by flx-sta** [comment]
Pull the latest beta branch to fix the failing tests

File: src/battle-scene.ts [hunk 1]
```diff
@@ -1975,46 +1981,77 @@ export default class BattleScene extends SceneBase {
...
+   * @param phase {@linkcode Phase} the phase to add
+   */
   unshiftPhase(phase: Phase): void {
     if (this.phaseQueuePrependSpliceIndex === -1) {
+      // .push() adds to end of array
       this.phaseQueuePrepend.push(phase);
     } else {
+      // .splice(index, num elements to remove, what to add)
+      // modifies array by inserting at index, removing num of elements after index
       this.phaseQueuePrepend.splice(this.phaseQueuePrependSpliceIndex, 0, phase);
```
Comment (torranx) [inline src/battle-scene.ts:2004]: These are all unnecessary comments - you're basically defining/describing JS array methods

File: src/battle-scene.ts [hunk 2]
```diff
@@ -1975,46 +1981,77 @@ export default class BattleScene extends SceneBase {
...
     if (this.phaseQueuePrepend.length) {
       while (this.phaseQueuePrepend.length) {
+        // appends phaseQueuePrepend to phaseQueue
+        // eg: phaseQueue = [4,5,6], phaseQUeuePrepend = [1,2,3]
+        // -> [1,2,3,4,5,6]
         this.phaseQueue.unshift(this.phaseQueuePrepend.pop());
       }
     }
+    // then starts from PhaseQueue, .shift() removes first elm of array
+    // populatePhaseQueue() adds a turnInit Phase at the end of phaseQueue (if the queue is emtpy)
```
Comment (torranx) [inline src/battle-scene.ts:2054]: ["Rule 1: Comments should not duplicate the code"](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/#h2-b7593c698a940)

File: src/field/pokemon.ts [hunk 3]
```diff
@@ -1995,11 +1995,13 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
...
-             * We explicitly require to ignore the faint phase here, as we want to show the messages
-             * about the critical hit and the super effective/not very effective messages before the faint phase.
-             */
+           * We explicitly require to ignore the faint phase here, as we want to show the messages
+           * about the critical hit and the super effective/not very effective messages before the faint phase.
+           */
           damage.value = this.damageAndUpdate(damage.value, result as DamageResult, isCritical, oneHitKo, oneHitKo, true);
           this.turnData.damageTaken += damage.value;
+
+          // queue critical message before effectiveness
```
Comment (torranx) [inline src/field/pokemon.ts:2004]: unnecessary comment

File: src/field/pokemon.ts [hunk 4]
```diff
@@ -2012,14 +2014,16 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
           source.turnData.damageDealt += damage.value;
           source.turnData.currDamageDealt = damage.value;
           this.battleData.hitCount++;
+          // create attack result and push to turnData
```
Comment (torranx) [inline src/field/pokemon.ts:2017]: unnecessary comment

File: src/phases.ts [hunk 5]
```diff
@@ -2662,6 +2667,7 @@ export class MovePhase extends BattlePhase {
       return false;
     });
 
+    // Readability?: this function declaration honestly gets in the way of readability of what start() is doing , move either to beginning or end
```
Comment (torranx) [inline src/phases.ts:2670]: unnecessary comment

File: src/phases.ts [hunk 6]
```diff
@@ -3025,8 +3031,11 @@ export class MoveEffectPhase extends PokemonPhase {
       if (--user.turnData.hitsLeft >= 1 && this.getTarget()?.isActive()) {
         this.scene.unshiftPhase(this.getNewHitPhase());
       } else {
+        // queue message for number of hits made by multi-move
+        // if multi-hit attack only hits once, still want to render a message
         const hitsTotal = user.turnData.hitCount - Math.max(user.turnData.hitsLeft, 0);
-        if (hitsTotal > 1) {
+        if (hitsTotal > 1 || user.turnData.hitsLeft > 0) {
+          // if there are multiple hits, or if there are hits of the multi-hit move left
```
Comment (torranx) [inline src/phases.ts:3038]: unnecessary comments

**Review by torranx — CHANGES_REQUESTED** [review]
I feel like a lot of the comments here were made for personal consumption when debugging the code. Also, please format the comments correctly. Another nit: instead of starting the sentence with "want...", using "We want..." is a lot better imo

File: src/battle-scene.ts (hunk 2 above)
Comment (DustinLin) [reply src/battle-scene.ts:2054]: will remove

**Comment by DustinLin** [comment]
Removed some of the comments you suggested, yes I agree we can make the assumption that people know what these functions do/can look them up themselves

**Pull request merged**